packages/reactive-element/src/decorators/query-assigned-nodes.ts

```diff
@@ -11,25 +11,69 @@
  * not an arrow function.
  */
 
-import {ReactiveElement} from '../reactive-element.js';
 import {decorateProperty} from './base.js';
 
+import type {ReactiveElement} from '../reactive-element.js';
+
+export interface QueryAssignedNodesOptions extends AssignedNodesOptions {
+  /**
+   * Name of the slot. Leave empty for the default slot.
+   */
+  slot?: string;
+  /**
+   * CSS selector used to filter the elements returned.
+   */
+  selector?: string;
```

Comment:
No strong opinion. One could also argue that this one should be the swiss army knife since the only reason `queryAssignedElements` really exists is so you don't have to do `selector: '*'` here.

Reply:
I would lean slightly towards not including it, preferring `@queryAssignedElements` for that case.

Reply:
I lean towards removing `selector` from `@queryAssignedNodes` option bag because it changes the type signature subtly, and `@queryAssignedElements` reads more clearly.

Reply:
I've removed and encouraged usage of `queryAssignedNodes` in the deprecated API documentation.